vllm/v1/worker/gpu_model_runner.py

```diff
@@ -944,6 +945,44 @@ def execute_model(
         sample_hidden_states = hidden_states[logits_indices]
         logits = self.model.compute_logits(sample_hidden_states, None)
 
+        # Apply guided decoding bitmasks if present
+        grammar_bitmask = scheduler_output.grammar_bitmask
+        if grammar_bitmask is not None:
+            grammar_bitmask = torch.from_numpy(grammar_bitmask)
+            # We receive the guided decoding bitmask from the scheduler, but the
+            # indices of the requests in the batch may not match the indices of
+            # the bitmask since the scheduler doesn't know how the gpu runner is
+            # ordering the requests in the batch. We need to sort the bitmask to
+            # match the order of the requests used here.
+            req_id_indices: Dict[str, int] = {}
+            indices_match = True
+            for req_id in self.input_batch.req_ids:
+                if req_id not in scheduler_output.guided_decoding_request_ids:
+                    # not a guided decoding request
+                    continue
+                batch_index = self.input_batch.req_id_to_index[req_id]
+                if batch_index != scheduler_output.guided_decoding_request_ids[
```

Comment:
nit: we don't need to search req_id twice in the dictionary.
```python
mask_index = scheduler_output.guided_decoding_request_ids.get(req_id)
if mask_index is None:
    continue
if batch_index != mask_index:
    ...
```

Reply:
done